src/pe/optional_header.rs

```diff
@@ -119,7 +119,7 @@ pub struct StandardFields {
     /// * For device drivers, this is the address of the initialization function.
     ///
     /// The entry point function is optional for DLLs. When no entry point is present, this member is zero.
-    pub address_of_entry_point: u64,
+    pub address_of_entry_point: u32,
```

Comment:
this seems like the only actually required fix, but even that is somewhat unclear to me? everything else seems to push the usize <-> u64/u32 casting down into the `pe::utils::find_offset`, which I admit does seem more elegant.

However, there was a certain clarity of passing a `usize` into these parse functions since they are ultimately used to compute 32/64-bit addresses/offsets.

Could you elaborate more on why this gets downgraded to a u32 though?

Reply:
Theoretically, this `u64` is _fine_. However,

1) Its 4 bytes half is always wasted because this is defined as `u32` as per PE specification[^1].

2) And as you mentioned, the `AddressOfEntryPoint` is always an RVA (`u32`), which is treated as `u32` as well, so consumers are forced to make unnecessary downcast if it is defined as `u64`.

[^1]: https://learn.microsoft.com/en-us/windows/win32/api/winnt/ns-winnt-image_optional_header64